jwst/pipeline/level2association.py

```diff
@@ -0,0 +1,100 @@
+"""Open a Level2 association given various inputs"""
```

Comment:
Because there is a touch of datamodel dependency in there, it seemed more appropriate for it to be more closely tied to the pipeline/model code. But, considering this is all JWST, the presumption that datamodels are baked into everything can be a given, allowing this to be considered a generic function.